nni/retiarii/graph.py

```diff
@@ -532,9 +536,12 @@ def __init__(self, graph, node_id, name, operation, _internal=False):
         # maybe we should copy it here or make Operation class immutable, in next release
         self.operation: Operation = operation
         self.label: Optional[str] = None
+        self.input_shape: Optional[list] = None
+        self.output_shape: Optional[list] = None
```

Comment:
Can we attach `input_shape` and `output_shape` onto `self.operation` as `self.operation.attrs['input_shape']`?

Reply:
I don't think `input_shape` and `output_shape` are operation's parameters as it could change along with different data size for the same operation. Actually this PR is to fix a bug in codegen. In codegen, all operation's parameter will be converted to the input argument when initializing the operation (https://github.com/microsoft/nni/blob/cb57730cd3f82b769f857c1e2f2f9a737084f7c5/nni/retiarii/operation.py#L136). This may cause the generated code has wrong input parameter.